src/components/flat-table/sort/sort.d.ts

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+export interface SortProps {
+   /** if `asc` it will show `sort_up` icon, if `desc` it will show `sort_down` */
+  sortType: 'asc' | 'desc' | false ;
+  /** Callback fired when the `FlatTableSortHeader` is clicked */
+  onClick: () => void;
```

Comment:
```suggestion
  onClick?: () => void;
```

Reply:
fixed in https://github.com/Sage/carbon/pull/2866/commits/ad5bc9949dfffb6c25d39f3041032cb87cf17e00